pkg/apps/blackduck/components/configmap/database/v1/configmap.go

```diff
@@ -1,61 +1,70 @@
 package v1
 
 import (
+	"fmt"
+	"reflect"
+	"strconv"
+
 	horizonapi "github.com/blackducksoftware/horizon/pkg/api"
 	"github.com/blackducksoftware/horizon/pkg/components"
 	blackduckapi "github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
-	"github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/types"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck"
 	"github.com/blackducksoftware/synopsys-operator/pkg/apps/store"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps/types"
 	apputils "github.com/blackducksoftware/synopsys-operator/pkg/apps/utils"
 	"github.com/blackducksoftware/synopsys-operator/pkg/protoform"
 	"github.com/blackducksoftware/synopsys-operator/pkg/util"
 	"k8s.io/client-go/kubernetes"
-	"strconv"
 )
 
+// BdConfigmap holds the Black Duck config map configuration
 type BdConfigmap struct {
 	config     *protoform.Config
 	kubeClient *kubernetes.Clientset
-	blackduck  *blackduckapi.Blackduck
+	blackDuck  *blackduckapi.Blackduck
 }
 
-func NewBdConfigmap(config *protoform.Config, kubeClient *kubernetes.Clientset, blackduck *blackduckapi.Blackduck) types.ConfigMapInterface {
-	return &BdConfigmap{config: config, kubeClient: kubeClient, blackduck: blackduck}
+// NewBdConfigmap returns the Black Duck config map configuration
+func NewBdConfigmap(config *protoform.Config, kubeClient *kubernetes.Clientset, blackDuck interface{}) (types.ConfigMapInterface, error) {
+	if reflect.TypeOf(blackDuck) != reflect.TypeOf(&blackduckapi.Blackduck{}) {
```

Comment:
No need to use reflect here.  Can just do:

db, ok := blackDuck.(*blackduckapi.Blackduck)

If !ok {
return err
}
return &DbConfigmap{....blackDuck:bd}

Reply:
fixed